v0.12.3 crashes with an ERR_INVALID_ARG_TYPE error when using NPM

### Description

sfdx-lwc-jest v0.12.3 crashes with the following error when running tests:
```
error TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received an instance of Object
```

#### Steps to Reproduce

Run these commands:

```sh
git clone git@github.com:pozil/sfdx-lwc-jest-repro.git
npm install
npm run test
```

#### Expected Results

All tests pass.

#### Actual Results

Test command crashes.

```sh
➜  sfdx-lwc-jest-repro git:(main) npm run test

> salesforce-app@1.0.0 test /Users/pozil/poz/git/sample-apps/sfdx-lwc-jest-repro
> npm run test:unit


> salesforce-app@1.0.0 test:unit /Users/pozil/poz/git/sample-apps/sfdx-lwc-jest-repro
> sfdx-lwc-jest

error TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received an instance of Object
```

### Version

-   @salesforce/sfdx-lwc-jest: 0.12.3
-   Node: v14.16.0

I cannot reproduce. When I run:

```
git clone git@github.com:trailheadapps/automation-components.git
cd automation-components
npm i
```

I get:

```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://nexus.soma.salesforce.com/nexus/content/groups/npm-all/@salesforce/eslint-plugin-lightning/-/eslint-plugin-lightning-0.1.1.tgz
npm ERR! 404
npm ERR! 404  '@salesforce/eslint-plugin-lightning@0.1.1' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'automation-components'
```

Does it work for you because it's cached on your local machine? If you run `npm cache clean --force` are you still able to run `npm i`?

@nolanlawson I confirm that the repo steps I provided are valid. I've cleaned my cache and I can still install and reproduce.

Looking at your logs, I see that you're using a private registry and this could be the issue that prevents your from installing:
> npm ERR! 404 Not Found - GET https://nexus.soma.salesforce.com/nexus/content/groups/npm-all/@salesforce/eslint-plugin-lightning/-/eslint-plugin-lightning-0.1.1.tgz

--- a/src/utils/test-runner.js
+++ b/src/utils/test-runner.js
@@ -32,7 +32,14 @@ function getJestPath() {
     const packageJsonPath = require.resolve('jest/package.json');
 
     const { bin } = require(packageJsonPath);
-    return path.resolve(path.dirname(packageJsonPath), bin);
+
+    // Account for difference in behavior between npm v6 and v7 (#221)
+    let normalizedBinPath = bin;
+    if (typeof bin === 'object') {
+        normalizedBinPath = bin.jest;
+    }
+
+    return path.resolve(path.dirname(packageJsonPath), normalizedBinPath);
 }
 
 function getJestArgs(argv) {

